src/Components/Components/src/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -722,6 +722,36 @@ private static int NextSiblingIndex(in RenderTreeFrame frame, int frameIndex)
             var valueChanged = !Equals(oldFrame.AttributeValueField, newFrame.AttributeValueField);
             if (valueChanged)
             {
+                // Whenever the original EventCallback or MulticastDelegate is different from the
```

Comment:
Before we do any of the following checks, can we check that `AttributeEventHandlerIdField` is nonzero? Otherwise I don't think we can assume that the attribute plays the role of an event handler even if its value type is a delegate.

Reply:
Yes I can. You mean when the delegate is actually executed to render the attribute value? Isn't it rendered here already by then? If not rendering, any example of a delegate that is not used as event handler?

I'll adapt this anyway, BTW.